mmdeploy/codebase/mmedit/models/contextual_attention.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+def _hw_shape(x):
+    return torch.gather(torch.tensor(x.shape), 0, torch.tensor(
+        (2, 3))).tolist()
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmedit.models.common.contextual_attention.ContextualAttentionModule.'
+    'patch_correlation')
+def contextual_attention__patch_correlation(ctx, self, x, kernel):
+    # Force tensor shape to avoid the following RuntimeError:
+    # Unsupported: ONNX export of convolution for kernel of unknown shape.
+    kernel += kernel.new_zeros(_hw_shape(kernel))
```

Comment:
As long as dynamic shape is not supported, reshape with a fix shape is cheaper than adding a tensor.

Reply:
I already have tried using `torch.Tensor.reshape`, but it didn't work. It caused RuntimeError as above comment.

Reply:
@grimoire Sorry, I made a mistake, it works.